components/service/glean/build.gradle

```diff
@@ -47,6 +47,15 @@ dependencies {
     implementation project(':concept-fetch')
     implementation project(':lib-fetch-httpurlconnection')
 
+    // We need a compileOnly dependency on the following block of testing
+    // libraries in order to expose the GleanTestRule to applications/libraries
+    // using the Glean SDK.
+    // We can't simply create a separate package otherwise we would need
+    // to provide a public API for the testing package to access the
+    // Glean internals, which is something we would not want to do.
```

Comment:
The only other workaround may be reflection ... but that has other downsides.

Reply:
Or the `RestrictTo.Scope` annotation which afaik still lets everyone call the method but the IDE may show a warning.

Reply:
We'd like not to go down the path of reflection again, if possible :( We also had bad luck with restricting things by showing warnings (using either `RestrictTo` or `VisibleForTesting` :-) ). I'm afraid we need something more... "opinionated" :)